### ISSUE TITLE
iOS support for transparent CAMetalLayer in WebGPU rendering

### ISSUE THREAD

**Issue by natedevxyz** [wcandillon/react-native-webgpu#141]
Would it be possible to add a feature specifically for iOS that allows developers to configure transparency in the CAMetalLayer used for WebGPU rendering?

I need to enable animations on top of other React Native views, but after testing multiple settings, I found that the underlying CAMetalLayer is opaque by default, preventing transparent WebGPU rendering.

The current clearValue used on iOS defaults to an opaque color, even if alpha values are set in the fragment shader:

```
const renderPassDescriptor: GPURenderPassDescriptor = {
      colorAttachments: [
        {
          view: undefined as unknown as GPUTextureView,
          clearValue: [1, 1, 1, 0], // Alpha is 0, but result is still opaque
          loadOp: 'clear' as const,
          storeOp: 'store' as const,
        },
      ],
    };
```

It seems like modifying the native settings below could address the issue:

Set self.opaque = NO; for the custom MetalView.
Set self.layer.opaque = NO; for the backing CAMetalLayer.

However, I’m not an expert so I'm unsure about the performance implications or if this is the ideal solution. If this isn't possible, are there any alternative approaches to achieve background transparency?

**Comment by wcandillon** [comment]
@natedevxyz I'm slowly starting to investigate transparency. Let me know if you have any thoughts on it for iOS. For Android our plan would be to use a shared hardware buffer texture. Maybe on iOS we can use what you suggested or maybe use shared texture IOSurfaces. I'm not up to date on this topic yet.

**Comment by natedevxyz** [comment]
@wcandillon To be honest, I ran into this issue after trying different settings to make the background transparent. After a while, I started wondering if there was some kind of native limitation and found a Stack Overflow post that confirmed my suspicion.

I’m not sure if what I suggested is the best way to go, otherwise I would have opened a PR. But if you're using the shared hardware buffer texture approach for Android, it could be beneficial to adopt a similar method on iOS to maintain consistency across platforms.

Thank you for all the work you’ve done on this!
